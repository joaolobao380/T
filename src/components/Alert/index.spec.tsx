import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import { Text } from 'react-native';

import { Alert } from '.';

describe('Alert Component', () => {
  it('Renders children and an info icon', () => {
    const testMessage = 'Test Alert Message';
    const { getByText } = render(
      <Alert>
        <Text>{testMessage}</Text>
      </Alert>
    );

    expect(getByText(testMessage)).toBeTruthy();
  });

  it('Should hides Alert on close button press', () => {
    const testMessage = 'Test Alert Message';
    const { getByTestId, queryByText } = render(
      <Alert>
        <Text>{testMessage}</Text>
      </Alert>
    );

    const closeButton = getByTestId('closeAlert');
    fireEvent.press(closeButton);

    expect(queryByText(testMessage)).toBeNull();
  });

  it('Renders correctly Alert', () => {
    const testMessage = 'Test Alert Message';
    const { toJSON } = render(
      <Alert>
        <Text>{testMessage}</Text>
      </Alert>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
