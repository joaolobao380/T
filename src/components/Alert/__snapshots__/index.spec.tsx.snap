// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Component Renders correctly Alert 1`] = `
<View
  style={
    {
      "backgroundColor": "#BAE6FC",
      "borderRadius": 8,
      "padding": 16,
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
        "position": "absolute",
        "right": 16,
        "top": 8,
      }
    }
    testID="closeAlert"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="20"
      bbWidth="20"
      fill="none"
      focusable={false}
      height={20}
      meetOrSlice={0}
      minX={0}
      minY={0}
      stroke="#444444"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "payload": 4282664004,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth="2"
      >
        <RNSVGPath
          d="M18 6 6 18"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4282664004,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        />
        <RNSVGPath
          d="m6 6 12 12"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4282664004,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 8,
        "marginRight": 20,
      }
    }
  >
    <View>
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="24"
        bbWidth="24"
        fill="none"
        focusable={false}
        height={24}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke="#444444"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4282664004,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        >
          <RNSVGCircle
            cx="12"
            cy="12"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            r="10"
            stroke={
              {
                "payload": 4282664004,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
          <RNSVGPath
            d="M12 16v-4"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4282664004,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
          <RNSVGPath
            d="M12 8h.01"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4282664004,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <Text>
        Test Alert Message
      </Text>
    </View>
  </View>
</View>
`;
